Short-Answer/Algorithms_Answers.md

```diff
@@ -2,14 +2,26 @@
 
 ## Exercise I
 
-a)
+a) It will be linear O(n) because it has while loop 
 
 
-b)
+b) It will be quadratic because there are two loops O(n^2)
```

Comment:
It totally looks like O(n^2) but the inner nested loop is doubling the value of j every time it runs. This means that the number of times the inner while loop runs is not growing at a linear rate - we would describe its runtime as O(log(n))! Finally, when we multiply the first loop we get O(n log(n)) :smiley: